pkg/component/certmanagement/deployment.go

```diff
@@ -0,0 +1,400 @@
+// SPDX-FileCopyrightText: SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package certmanagement
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	operatorv1alpha1 "github.com/gardener/gardener/pkg/apis/operator/v1alpha1"
+	resourcesv1alpha1 "github.com/gardener/gardener/pkg/apis/resources/v1alpha1"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/component"
+	"github.com/gardener/gardener/pkg/resourcemanager/controller/garbagecollector/references"
+	gardenerutils "github.com/gardener/gardener/pkg/utils"
+	kubernetesutils "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"github.com/gardener/gardener/pkg/utils/managedresources"
+)
+
+const (
+	// DeploymentManagedResourceName is the name of the managed resource for the resources.
+	DeploymentManagedResourceName = "cert-management-deployment"
+
+	resourceName  = "cert-controller-manager"
+	containerName = "cert-management"
+
+	serverPort        = 8080
+	rsaPrivateKeySize = 3072
+)
+
+//go:embed assets/cert.gardener.cloud_certificaterevocations.yaml
+var crdRevocations string
+
+//go:embed assets/cert.gardener.cloud_certificates.yaml
+var crdCertificates string
+
+//go:embed assets/cert.gardener.cloud_issuers.yaml
+var crdIssuers string
+
+// newCertManagementDeployment creates a new instance of DeployWaiter for the CertManagement deployment.
+func newCertManagementDeployment(cl client.Client, values Values) component.DeployWaiter {
+	return &certManagementDeployment{
+		client:     cl,
+		namespace:  values.Namespace,
+		image:      values.Image,
+		deployment: values.Deployment,
+	}
+}
+
+type certManagementDeployment struct {
+	client     client.Client
+	namespace  string
+	image      string
+	deployment *operatorv1alpha1.CertManagementDeployment
+}
+
+func (d *certManagementDeployment) Deploy(ctx context.Context) error {
+	if err := d.deploy(ctx); err != nil {
+		return err
+	}
+
+	// MIGRATION-FROM-LSS
+	return d.deleteHelmRelease(ctx)
```

Comment:
We probably don't need this migration anymore, do we?

Reply:
removed, thanks for catching!